crates/core/tedge/src/cli/config/commands/get.rs

```diff
@@ -19,6 +19,7 @@ impl Command for GetConfigCommand {
             }
             Err(tedge_config::ReadError::ConfigNotSet { .. }) => {
                 eprintln!("The provided config key: '{}' is not set", self.key);
+                std::process::exit(1)
             }
             Err(tedge_config::ReadError::ReadOnlyNotFound { message, key }) => {
                 eprintln!("The provided config key: '{key}' is not configured: {message}",);
```

Comment:
```suggestion
                eprintln!("The provided config key: '{key}' is not configured: {message}",);
                std::process::exit(1)
```

Even better: simply remove the case for `Err(tedge_config::ReadError::ReadOnlyNotFound {})` and let it handled by the catch all case .i.e. `Err(err) => return Err(err.into())`.

Reply:
The default of `clap` returns `2` for the error case. Your suggestion will change the exit code

Reply:
I'm speaking about the second case `ReadOnlyNotFound`, not the one you fixed `ConfigNotSet`.

`ReadOnlyNotFound` must exit 1 or 2 but not 0 as today.

```
$ tedge config unset device.cert_path
$ tedge config get device.id                                                   
The provided config key: 'device.id' is not configured: The device id is read from the device certificate.
To set 'device.id' to some <id>, you can use `tedge cert create --device-id <id>`.
$ echo $?         
0
```

Reply:
This `execute()` is called here. 

https://github.com/thin-edge/thin-edge.io/blob/e77a5dfb64e7dd9da45b7ba2429af2d56101ba52/crates/core/tedge/src/main.rs#L65-L66

> Even better: simply remove the case for Err(tedge_config::ReadError::ReadOnlyNotFound {}) and let it handled by the catch all case .i.e. Err(err) => return Err(err.into()).

I prefer your original code suggestion, just to add `std::process::exit(1)` to `ReadOnlyNotFound` case. The reason is, if it returns `Err`, `.with_context()` will be called, that adds extra stderr output compared to now. I feel it's unnecessary change to user.

Reply:
Done in 0458c17271a50a6b6efb9dd9167dc5a68eeb4083